gax/src/main/java/com/google/api/gax/rpc/ApiException.java

```diff
@@ -30,11 +30,13 @@
 package com.google.api.gax.rpc;
 
 import com.google.common.base.Preconditions;
+import java.util.Map;
 
 /** Represents an exception thrown during an RPC call. */
 public class ApiException extends RuntimeException {
   private static final long serialVersionUID = -4375114339928877996L;
 
+  private ErrorDetails errorDetails;
```

Comment:
Looks like this can be final

Reply:
If a field is not modifiable it must be final. Making final everyitng which is actually final also simplifies greatly any concurrent code for example. Here it seems like the cost of making it final would be simply just making it null in two other constructors, which is super cheap.

Reply:
Made it final as suggested.